File: inbound/backend/src/Models/Kafka/ReceptionCreated.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace backend.Models.Kafka;

public class ReceptionCreated
{
    public ReceptionCreated() { }

    public ReceptionCreated(int asnId)
    {
        AsnId = asnId;
    }

    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [ForeignKey("Asn")]
    public int? AsnId { get; set; }

    [Required]
    [JsonPropertyName("asn")]
    public ASN? Asn { get; set; }

}

File: inbound/frontend/lib/model/entities/reception_created.dart
import 'package:frontend/model/entities/asn.dart';

class ReceptionCreated {
  int id;
  int asnId;
  Asn asn;

  ReceptionCreated({required this.id, required this.asnId, required this.asn});

  ReceptionCreated.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        asnId = json['asnId'],
        asn = Asn.fromJson(json['asn']);
}
